engine/engine.py

```diff
@@ -18,6 +19,7 @@ class DreamTexturesRenderEngine(bpy.types.RenderEngine):
     bl_label = "Dream Textures"
     bl_use_preview = False
     bl_use_postprocess = True
+    bl_use_gpu_context = actor.main_thread_rendering
```

Comment:
For some reason `bl_use_gpu_context = True` allows for rendering annotations in background mode, but freezes under normal use. And `bl_use_gpu_context = False` works fine under normal use, but freezes while in background mode. So `bl_use_gpu_context = actor.main_thread_rendering` is the best that I'll get this to work.